validator/server_arb/boldmach/machine.go

```diff
@@ -0,0 +1,151 @@
+package boldmach
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/offchainlabs/nitro/validator"
+	"github.com/offchainlabs/nitro/validator/server_arb"
+)
+
+// boldMachine wraps a server_arb.MachineInterface.
+type boldMachine struct {
+	inner       server_arb.MachineInterface
+	zeroMachine *server_arb.ArbitratorMachine
+	hasStepped  bool
+}
+
+// Ensure boldMachine implements server_arb.MachineInterface.
+var _ server_arb.MachineInterface = (*boldMachine)(nil)
+
+func newBoldMachine(inner server_arb.MachineInterface) *boldMachine {
+	z := server_arb.NewFinishedMachine(inner.GetGlobalState())
+	return &boldMachine{
+		inner:       inner,
+		zeroMachine: z,
+		hasStepped:  false,
+	}
+}
+
+// MachineWrapper wraps a server_arb.MachineInterface and adds one step to the
+// front of the machine's execution.
+//
+// This zeroth step should be at the same global state as the inner arbitrator
+// machine has at step 0, but the machine is in the Finished state rather than
+// the Running state.
+func MachineWrapper(inner server_arb.MachineInterface) server_arb.MachineInterface {
```

Comment:
It seems like this function is only used in tests, so in production this wouldn't work as it's missing the zero step this bold machine wrapper adds. This also highlights an issue with the current approach: to make upgrading smooth, the same nitro-val instance should work pre-bold and post-bold. We'll likely need to pass a flag in createExecutionRun indicating if a bold machine should be created. I'll also note we don't need that flag for normal block validation, which is the same between pre- and post-bold.